src/openfermion/ops/_binary_operator.py

```diff
@@ -0,0 +1,494 @@
+# symbolic binary class for decoder definitions (arxiv 1712.07067)
+
+import copy
+import numpy as np
+
+
+class SymbolicBinaryError(Exception):
+    pass
+
+
+class SymbolicBinary(object):
+    """
+        SymbolicBinaries are non-linear binary functions that are expressed analytically, they are
+        components of decoders that are used to rotate terms in qubit basis to fermionic basis. 
+        
+        A SymbolicBinary represents a sum of terms. Each term can consist of two types of binary variables:
+        (n,'W') where n stands for the qubit index, or (1,'1') for constant 1.
+         
+        The SymbolicBinary class overloads operations for manipulation of binary functions obeying binary rules.
```

Comment:
Please run the pep8 linter. This line is over 100 characters. We're old school - keep to 80 char per line please.